Added navigation and AppBar in Pokemon Details

kamil-matula opened the pull request:
1. Refactored project structure to separate main screens from details screens.
2. Added three new screens for Pokemon Details.
3. Added Bottom Navigation for Pokemon Details.
4. Added Custom App Bar with sprite and button for Pokemon Details - the data to display is passed through the parent screen.

Display:
![image](https://user-images.githubusercontent.com/43967269/116380805-9bb3db80-a814-11eb-9628-554f9e71db06.png)

Unfortunately, I don't know how to pass the data from main screen to details screen (and from details tabs to one of the details screen), so it will have to be done in next commit / next task.

kamil-matula committed:
Added navigation and AppBar in Pokemon Details

1. Refactored project structure to separate main screens from details screens.
2. Added three new screens for Pokemon Details.
3. Added Bottom Navigation for Pokemon Details.
4. Added Custom App Bar with sprite and button for Pokemon Details - the data to display is passed through the parent screen.

piotr-wawro commented on App.js:
```diff
@@ -12,9 +12,10 @@ export default function App() {
     <PaperProvider theme={CustomTheme}>
       <NavigationContainer>
-        <Stack.Navigator headerMode='none' initialRouteName='Home'>
+        <Stack.Navigator headerMode='none' initialRouteName='Details'>
           <Stack.Screen name='ApiTesting' component={ApiTesting} />
           <Stack.Screen name='Home' component={HomeTabs} />
+          <Stack.Screen name='Details' component={DetailsTabs} />
```
Isn't that temporary change? It looks like.

kamil-matula replied:
Yes, it is. For now launching the app results in showing Details Screen instead of Main Screen, because it is new feature. It will be changed after implementing navigation between these two screens. Probably the ApiTesting screen will be also removed from the Stack.

Blazevarjo replied:
I think's that redundant but that's okay

piotr-wawro commented on src/components/DetailsAppBar.js:
```diff
@@ -0,0 +1,49 @@
+      <View style={styles.leftCorner}>
+        <Button
+          icon='arrow-left'
+          color='#F2F2F2'
```
Add color to theme

Blazevarjo commented on App.js:
```diff
@@ -3,7 +3,7 @@ import { Provider as PaperProvider } from 'react-native-paper';
 import { createStackNavigator } from '@react-navigation/stack';
 
-import { ApiTesting, HomeTabs } from './src/screens';
+import { ApiTesting, HomeTabs, DetailsTabs } from './src/screens/index';
```
index is unnecessary

kamil-matula committed:
Applied suggestions to DetailsScreen Update

1. Exported color string to theme file.
2. Simplified one of the imports.
3. Set 'Home' route as default.

Blazevarjo committed:
Merge branch 'main' into DetailsScreen
